lib/rubocop/cop/lint/mixed_regexp_capture_types.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Do not mix named captures and numbered captures in a Regexp literal
+      # because numbered capture is ignored if they're mixed.
+      # Replace numbered captures with non-capturing groupings or
+      # named captures.
+      #
+      #   # bad
+      #   /(?<foo>FOO)(BAR)/
+      #
+      #   # good
+      #   /(?<foo>FOO)(?<bar>BAR)/
```

Comment:
I think it's even more helpful to show an example of numbered capture for users.

```ruby
      #   # good
      #   /(FOO)(BAR)/
```

Reply:
Good point, thanks!
I added it as a good example. d8144b88883dbe4b28edaff347fa2c3ab87d2334

I think RuboCop prefers named capture than numbered capture. For example, we have a rule in the style guide. https://github.com/rubocop-hq/ruby-style-guide#no-numbered-regexes
So I added the suggested example to the bottom of the examples. 
It is not bad, and I prefer it with a simple Regexp, but named capture is more descriptive.

Reply:
I get it! Thanks for the explanation.